library/src/main/java/com/novoda/downloadmanager/lib/BatchCompletionBroadcaster.java

```diff
@@ -0,0 +1,57 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.support.annotation.NonNull;
+import android.support.v4.content.LocalBroadcastManager;
+
+public class BatchCompletionBroadcaster {
+
+    static final String ACTION_BATCH_COMPLETE = "com.novoda.downloadmanager.BATCH_COMPLETE";
+    static final String EXTRA_BATCH_ID = "com.novoda.downloadmanager.EXTRA.batch_id";
+
+    private final LocalBroadcastManager localBroadcastManager;
+    private final String packageName;
+    private BatchCompletionListener batchCompletionListener;
+
+    public static BatchCompletionBroadcaster newInstance(Context context) {
+        return new BatchCompletionBroadcaster(LocalBroadcastManager.getInstance(context), context.getApplicationContext().getPackageName());
+    }
+
+    BatchCompletionBroadcaster(LocalBroadcastManager localBroadcastManager, String packageName) {
+        this.localBroadcastManager = localBroadcastManager;
+        this.packageName = packageName;
+    }
+
+    public void notifyBatchCompletedFor(long batchId) {
+        Intent intent = new Intent(ACTION_BATCH_COMPLETE);
+        intent.setPackage(packageName);
+        intent.putExtra(EXTRA_BATCH_ID, batchId);
+        localBroadcastManager.sendBroadcast(intent);
+    }
+
+    public void registerReceiver(BatchCompletionListener batchCompletionListener) {
```

Comment:
since this is a library it'd be better if you hold a list of `BatchCompletionListener`, then register would add to the list and unregister would unregister one in particular

take this as example [register](http://developer.android.com/reference/android/content/Context.html#registerReceiver(android.content.BroadcastReceiver, android.content.IntentFilter)) and [unregister](http://developer.android.com/reference/android/content/Context.html#unregisterReceiver(android.content.BroadcastReceiver))